src/power/battery.rs

```diff
@@ -0,0 +1,177 @@
+use crate::create_issue;
+use crate::power::get_best_path;
+use crate::Error;
+use colored::Colorize;
+use std::any::Any;
+use std::fs;
+use std::fs::read_dir;
+use std::path::Path;
+
+const BATTERY_CHARGE_PATH: [&str; 4] = [
+    "/sys/class/power_supply/BAT/",
+    "/sys/class/power_supply/BAT0/",
+    "/sys/class/power_supply/BAT1/",
+    "/sys/class/power_supply/BAT2/",
+];
+
+pub fn has_battery() -> bool {
+    let power_dir = Path::new("/sys/class/power_supply/");
+    let dir_count = read_dir(power_dir).into_iter().len();
+    dir_count > 0
+}
+
+pub enum BatteryConditionType {
+    Energy,
+    Charge,
+    None,
+}
+
+pub enum BatteryStatus {
+    Charging,
+    Discharging,
+    Full,
+    Unknown,
+}
+
+pub struct Battery {
+    pub sys_parent_path: String,
+    pub capacity: i8,
+    pub condition_type: BatteryConditionType,
+    pub condition: i8,
+    pub charge_full: i32,
+    pub charge_full_design: i32,
+    pub energy_full: i32,
+    pub energy_full_design: i32,
+    pub status: BatteryStatus,
+}
+
+impl Battery {
+    pub fn new() -> Result<Battery, Error> {
+        let mut obj = Battery {
+            sys_parent_path: "unknown".to_string(),
+            capacity: 0_i8,
+            condition_type: BatteryConditionType::None,
+            condition: 0_i8,
+            charge_full: 0_i32,
+            charge_full_design: 0_i32,
+            energy_full: 0_i32,
+            energy_full_design: 0_i32,
+            status: BatteryStatus::Unknown,
+        };
+        let path: &str = match get_best_path(BATTERY_CHARGE_PATH) {
+            Ok(path) => path,
+            Err(error) => {
+                if error.type_id() == Error::IO.type_id() {
+                    // Make sure to return IO error if one occurs
+                    return Err(error);
+                }
+                // If it doesn't exist then it is plugged in so make it 100% percent capacity
+                eprintln!("We could not detect your battery.");
+                create_issue!("If you are on a laptop");
+                return Err(Error::Unknown);
+            }
+        };
+        obj.sys_parent_path = path.to_string();
+        obj.check_condition_type();
+        Ok(obj)
+    }
+
+    fn read_charge(&mut self) -> Result<(), Error> {
+        let charge_path = self.sys_parent_path.to_string() + "capacity";
+        let mut cap_str = fs::read_to_string(charge_path)?;
+
+        // Remove the \n char
+        cap_str.pop();
+
+        let charge = cap_str.parse::<i8>().unwrap();
+        self.capacity = charge;
+
+        Ok(())
+    }
+
+    // TODO: Move this to display.rs
```

Comment:
Would this make sense to be I display.rs?

Reply:
* in

Reply:
display.rs handles displaying things like power, turbo, cpus, etc. I think its fair for it to deal with printing battery info,